js/app/src/lite/index.ts

```diff
@@ -116,55 +118,104 @@ export function create(options: Options): GradioAppController {
 	};
 
 	let app: SvelteComponent;
+	let app_props: any;
 	function showError(error: Error): void {
 		if (app != null) {
 			app.$destroy();
 		}
-
-		app = new ErrorDisplay({
-			target: options.target,
-			props: {
-				is_embed: !options.isEmbed,
-				error
-			}
-		});
+		if (options.playground) {
+			app = new Playground({
+				target: options.target,
+				props: {
+					...app_props,
+					code: options.code,
+					error_display: {
+						is_embed: !options.isEmbed,
+						error
+					},
+					loaded: true
+				}
+			});
+			app.$on("code", (code) => {
+				options.code = code.detail.code;
+				worker_proxy
+					.runPythonCode(options.code)
+					.then(launchNewApp(true))
+					.catch((e) => {
+						showError(e);
+						throw e;
+					});
+			});
```

Comment:
What if we move this snippet to the `<Playground />` component and call `worker_proxy.runPythonCode(...)` in `<Playground />` directly?
The logic around code updates will be encapsulated into `<Playground />` and we won't need to dispatch and catch the `code` event across the components.

Reply:
That's what I intended to do initially but couldn't figure out how. I can runPythonCode inside <Playground /> but I can't launchNewApp afterwards (since the whole playground is the app). running the code just kills the port. How do I work around this?

Reply:
Ah sorry you are right, I forgot about `launchNewApp`.
Maybe we can fix `lite/index.ts` and `Playground.svelte` so that `<Playground />` is kept mounted when the code is changed and only `<Index />` is refreshed inside it, but the current code should be good for now.